src/Mod/Part/Gui/ViewProviderExt.cpp

```diff
@@ -599,69 +625,89 @@ std::vector<Base::Vector3d> ViewProviderPartExt::getSelectionShape(const char* /
     return {};
 }
 
-void ViewProviderPartExt::setHighlightedFaces(const std::vector<App::Color>& colors)
+void ViewProviderPartExt::setHighlightedFaces(const std::vector<App::Material>& materials)
 {

```

Comment:
What's the reason of dropping the code block below? The TouchOnColorChange thing has been added with 75e639c94c3c7a9 which is used by object links.

The SoUpdateVBOAction is needed to notify the SoBrepFaceSet about colour changes. In case the viewer uses VBO (Vertex Buffer Object) this will break the rendering.